kie-internal/src/main/java/org/kie/internal/task/api/prediction/PredictionOutcome.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A class to encapsulate prediction results from a {@link org.kie.internal.task.api.prediction.PredictionService}
+ * implementation.
+ */
+package org.kie.internal.task.api.prediction;
+
+import java.util.Map;
+
+/**
+ * Encapsulates results from a {@link PredictionService}.
+ */
+public class PredictionOutcome {
+
+    private boolean present;
+
+    private double confidenceLevel;
+    
+    private double confidenceThreshold;
+
+    private Map<String, Object> data;
+
+    /**
+     * Creates an empty prediction.
+     */
+    public PredictionOutcome() {
+        this.present = false;
```

Comment:
An empty prediction has undefined confidenceLevel and confidenceThreshold, this may cause unpredictable result at isCertain method.

Reply:
I don't think this would be unpredictable, as if these two values are empty present is equal to false so isCertain() is false (which is correct). If present is true then it must have been init using the constructor with parameters, therefore confidenceLevel and confidenceThreshold will have been init. Can you advice if you agree or if you wish confidenceThreshold and confidenceLevel to be init to 0.0 at the top of the class please.

Reply:
@rebeccaSimmonds19 You're right, so empty prediction will always return false at isCertain(). No need initialization therefore for these attributes. Just add a comment to the isCertain method: "Returns _true_ whether a prediction has a confidence level above the specified threshold _or false, otherwise or for empty predictions_", wdyt?

Reply:
sounds good @gmunozfe I will make those changes this morning :)